### PR TITLE
Bug/dwr 728 leak student data

### PR DESCRIPTION
**Pull request by bgsavage**
Now throwing when no exams are found on the get student exams call.  This solution seems simpler than implementing all the sql permission checks in the studentRepository.findOneById if we are already making these same checks in the exam lookup anyway.

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/student/DefaultStudentExamService.java [hunk 1]
```diff
@@ -37,6 +37,10 @@ public StudentExamHistory getExams(@NotNull final User user, final long studentI
                 student.getId()
         );
 
+        if(exams.size() == 0) {
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/student/DefaultStudentExamService.java]: nit: use ```isEmpty()```

Comment (malaffoon) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/student/DefaultStudentExamService.java]: +1

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/student/ReportingExamItemService.java [hunk 2]
```diff
@@ -28,10 +28,13 @@ public ReportingExamItemService(final ExamItemRepository examItemRepository,
 
     @Override
     public GroupExamItemScores getExamItems(@NotNull final User user, final long examId) {
-        final PermissionScope scope = user.getPermissionsById().get("GROUP_PII_READ").getScope();
-
         final List<ExamItem> examItems = examItemRepository.findAllByExamId(user.getPermissionsById(), user.getGroupsById().keySet(), examId);
         final List<AssessmentItem> assessmentItems = assessmentItemRepository.findAllForExam(examId);
+
+        if(examItems.isEmpty() || assessmentItems.isEmpty()) {
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/student/ReportingExamItemService.java]: you could optimize this by moving this block before the assessment item lookup. if there are no exam items we dont need to lookup the assessment items

Comment (bgsavage) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/student/ReportingExamItemService.java]: Ideally these two calls could be made asynchronously as they don't depend on each other.  

But I can optimize as you suggest for now.

**Pull request merged**